utils/analyticsmetrics_test.go

```diff
@@ -0,0 +1,160 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	coretests "github.com/jfrog/jfrog-cli-core/v2/common/tests"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	xscservices "github.com/jfrog/jfrog-client-go/xsc/services"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	testMsi                             = "27e175b8-e525-11ee-842b-7aa2c69b8f1f"
+	lowerAnalyticsMetricsMinXscVersion  = "1.6.0"
+	higherAnalyticsMetricsMinXscVersion = "1.10.0"
+)
+
+func TestCalcShouldReportEvents(t *testing.T) {
+	// Save original environment information.
+	originalJfMsi := os.Getenv(jfMsiEnvVariable)
+	defer func() {
+		err := os.Setenv(jfMsiEnvVariable, originalJfMsi)
+		assert.NoError(t, err)
+	}()
+	originalReportUsage := os.Getenv(coreutils.ReportUsage)
+	defer func() {
+		err := os.Setenv(coreutils.ReportUsage, originalReportUsage)
+		assert.NoError(t, err)
+	}()
+
+	err := os.Setenv(coreutils.ReportUsage, "")
+	assert.NoError(t, err)
+	err = os.Setenv(jfMsiEnvVariable, "")
+	assert.NoError(t, err)
```

Comment:
```suggestion
	cleanReportEnv := clientTests.SetEnvWithCallbackAndAssert(t, coreutils.ReportUsage, "")
    cleanMSIEnv := clientTests.SetEnvWithCallbackAndAssert(t, jfMsiEnvVariable, "")
```

Use clientTests.SetEnvWithCallbackAndAssert